csharp/src/Microsoft.ML.OnnxRuntime/ManagedProjections.shared.cs

```diff
@@ -225,53 +158,31 @@ private OrtValue CreateMapProjection(NamedOnnxValue node, NodeMetadata elementMe
         /// <param name="disposables">cleanup list</param>
         /// <returns></returns>
         /// <exception cref="OnnxRuntimeException"></exception>
-        private OrtValue CreateTensorProjection(NamedOnnxValue node, NodeMetadata elementMeta, DisposableList<IDisposable> disposables)
+        private static OrtValue CreateTensorProjection(NamedOnnxValue node, NodeMetadata elementMeta)
         {
-            var ortValue = OrtValue.CreateFromTensorObject(node.Value,
-                out MemoryHandle? memoryHandle, out TensorElementType elementType);
-            disposables.Add(ortValue);
-
-            if (memoryHandle.HasValue)
-            {
-                disposables.Add(memoryHandle);
-            }
-
-            if (elementType != elementMeta.ElementDataType)
+            if (!(node.Value is TensorBase))
             {
                 throw new OnnxRuntimeException(ErrorCode.InvalidArgument,
-                    $"Tensor element data type discovered: {elementType} metadata expected: {elementMeta.ElementDataType}");
+                    $"NamedOnnxValue contains: {node.Value.GetType()}, expecting a Tensor<T>");
             }
 
-            return ortValue;
-        }
-
-        #region IDisposable
-        /// <summary>
-        /// IDisposable implementation
-        /// </summary>
-        /// <param name="disposing">true if invoked by Dispose()</param>
-        protected virtual void Dispose(bool disposing)
-        {
-            if (_disposed)
+            OrtValue ortValue = OrtValue.CreateFromTensorObject(node.Value as TensorBase, out TensorElementType elementType);
+            try 
             {
-                return;
+                if (elementType != elementMeta.ElementDataType)
+                {
+                    throw new OnnxRuntimeException(ErrorCode.InvalidArgument,
+                        $"Tensor element data type discovered: {elementType} metadata expected: {elementMeta.ElementDataType}");
+                }
             }
-
-            // dispose managed state (managed objects).
-            if (disposing)
+            catch(Exception)
             {
-                _disposables.Dispose();
+                ortValue.Dispose();
+                throw;
             }
```

Comment:
Can we use `finally` for the Dispose instead of catch/rethrow?

Reply:
Not here. ortValue is a result, we should return it on success.